examples/data-objects/table-document/.eslintrc.js

```diff
@@ -11,4 +11,16 @@ module.exports = {
 	rules: {
 		"@typescript-eslint/strict-boolean-expressions": "off",
 	},
+	overrides: [
+		{
+			// Rules only for test files
+			files: ["*.spec.ts", "src/test/**"],
+			rules: {
+				// ESLint's resolver doesn't resolve relative imports of ESNext modules correctly, since
+				// it resolves the path relative to the .ts file (and assumes a file with a .js extension
+				// should exist there)
+				"import/no-unresolved": ["error", { ignore: ["^\\.(.*)\\.js$"] }],
```

Comment:
i'd love to make eslint's module resolver understand these imports, but I'm not sure it's possible out of the box (we'd have to maintain our own resolver, and I'm not sure if it would be a trivial wrapper over the base node resolver or not). I don't think this rule provides particularly high value for relative imports though, so IMO it's not the worst.

Reply:
Someone must have written a solution for this already, right? Typescript transpiled to a different folder + ESM should be a common scenario by now...? Can be addressed separately.

Reply:
thanks for prodding me to look into this again. I think we can actually make this work nicely by removing [eslint-plugin-unused-imports](https://www.npmjs.com/package/eslint-plugin-unused-imports), transitioning to [eslint-plugin-import](https://www.npmjs.com/package/eslint-plugin-import), and updating our import resolve to be [eslint-import-resolver-typescript](https://www.npmjs.com/package/eslint-import-resolver-typescript) (where the default `extensionAlias` should work perfectly). that'll require a release of the common lint config, though.

Reply:
[AB#4614](https://dev.azure.com/fluidframework/internal/_workitems/edit/4614)